NickvisionTubeConverter.Shared/Models/DownloadHistoryItem.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+
+namespace NickvisionTubeConverter.Shared.Models;
+
+/// <summary>
+/// Item in download history
+/// </summary>
+public class DownloadHistoryItem
+{
```

Comment:
We should have a `public string Url { get; set; }` property and a Constructor such as this
```csharp
public DownloadHistoryItem(string url)
{
    Url = url;
    Title = "";
    Date = DateTime.Now;
    Path = "";
}
```

Reply:
what's the point of keeping URL in HistoryItem if it's already used as the key in history dictionary?

Reply:
Because for the sake of object design, this object doesn't necessarily need to be used in a dictionary. Therefore the Url property will become the only way to differentiate the objects.

Reply:
and we can even clean up the config file and just use `List<DownloadHistoryItem>` but the dictionary ensures we have a single `DownloadHistoryItem` per Url. I might be able to change it into a `Set<DownloadHistoryItem>` will look into it.

Reply:
> and we can even clean up the config file and just use List<DownloadHistoryItem> but the dictionary ensures we have a single DownloadHistoryItem per Url. I might be able to change it into a Set<DownloadHistoryItem> will look into it.

seems like Dictionary is the best structure to use for the case of keeping one recent history item per url. But please still add the Url property and new constructor for API design...

similar to how in Denaro we have

```csharp
class Transaction 
{
     public uint ID { get; init; }
     
     public Transaction(uint id) { ... }
}

public Dictionary <uint, Transaction> Transactions ...
```
The only thing is for `DownloadHistoryItem` `Url` needs to have `set` for JsonSerialization and Deserialization to work.